README.md

```diff
@@ -1,2 +1,358 @@
-# node-bitdotio
-Node SDK for bit.io
+# bit.io Node.js SDK
+
+Node.js SDK for connecting to [bit.io](https://bit.io) databases and interacting with
+the [bit.io](https://bit.io) developer API.
+
+## Installation
+
+```sh
+npm install node-bitdotio
+```
+
+## Usage
+
+The `node-bitdotio` package consists of a `bitdotio` SDK object which provides helpful
+utilities for creating and managing pre-configured connections to bit.io databases
+via [`node-postgres`](https://node-postgres.com/), as well as easy methods for accessing
+the functionality exposed by the [bit.io developer API](https://docs.bit.io/reference).
+
+Once you have `node-bitdotio` installed all you need is your API key to start working
+with bit.io.
+
+You can get your API key by logging into bit.io and opening [the "Connect" tab](https://docs.bit.io/docs/your-connection-credentials)
+on a database page.
+
+### Example usage
+
+See API reference at the bottom of this document for a full list of methods provided by
+the SDK.
+
+```js
+const { createReadStream } = require('fs');
+const bitdotio = require('node-bitdotio');
+
+const BITIO_KEY = process.env.BITIO_KEY;
+
+// Instantiate the bitdotio SDK
+const b = bitdotio(BITIO_KEY);
+
+// Create a database
+b.createDatabase({ name: "demo-db" }).then(async (dbMetadata) => {
+  const dbName = dbMetadata.name;
+
+  // Create a connection pool for our newly created database
+  const pool = b.getPool(dbName);
+
+  // Connect to the database using the pool and insert some data
+  await pool.connect().then(async (client) => {
+    await client.query("CREATE TABLE test (foo integer, bar text)");
+    const data = [[1, "one"], [2, "two"], [3, "three"]];
+    for (const row of data) {
+      await client.query("INSERT INTO test (foo, bar) VALUES ($1, $2)", row);
+    }
+    // Make sure to return all clients acquired from a connection pool
+    client.release();
+  });
+
+  // Run a query against our new table using the API
+  await b.query(dbName, "SELECT * FROM test", "objects").then((results) => {
+    console.table(results.data);
+  });
+
+  const sleep = (ms) => new Promise((resolve) => { setTimeout(resolve, ms); });
+
+  // Start a file upload job and wait for it to complete
+  const file = createReadStream('./cities.csv');
+  let jobStatus = await b.createImportJob(dbName, { tableName: "cities", type: "file", file });
+  while (true) {
+    jobStatus = await b.getImportJob(jobStatus.id);
+    if (jobStatus.state === "DONE") {
+      break;
+    }
+    if (jobStatus.state === "FAILED") {
+      const { errorId, errorType, errorDetails } = jobStatus;
+      throw new Error(
+        `Import job failed. Error ID: ${errorId}, Error Type: ${errorType}, Error Details: ${errorDetails}`
+      );
+    }
+
+    // Sleep a bit to not overload the API
+    await sleep(200);
+  }
+});
+```
+
+### Making queries
+
```

Comment:
Add an opening paragraph enumerating the two options + a 1 sentence tradeoff/indicating which is recommended (possible in the header as well)

Reply:
Done